daisy/step_create_instances.go

```diff
@@ -17,22 +17,43 @@ package daisy
 import (
 	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
 	"path"
 	"sync"
 	"time"
 )
 
 // CreateInstances is a Daisy CreateInstances workflow step.
-type CreateInstances []*Instance
+type CreateInstances struct {
+	Instances     []*Instance
+	InstancesBeta []*InstanceBeta
+}
+
+// UnmarshalJSON unmarshals Instance.
+func (ci *CreateInstances) UnmarshalJSON(b []byte) error {
+	var instancesBeta []*InstanceBeta
+	if err := json.Unmarshal(b, &instancesBeta); err != nil {
+		return err
```

Comment:
A bit confusion to me. It looks like if the json is compatible with both beta and GA, it will be initialized twice, and exist in both ci.InstancesBeta and ci.Instances. Is that the case? If so, will it be created twice?

Reply:
Yes. The same logic is used in CreateImages. validate/run methods in StepCreateInstances used instanceUsesBetaFeatures() to determine which one to use.